### PR TITLE
Port to Microsoft Edge 

### PR DESCRIPTION
**Pull request by Jack-Q**
### Description

A port of this extension to Microsoft Edge.

This is port contains some modifications to the original extension to some compatibility issues in Microsoft Edge.  #480

### CODE REVIEW

File: src/octotree.js [hunk 1]
```diff
@@ -190,4 +190,10 @@ $(document).ready(() => {
       return $toggler.is(':visible')
     }
   }
-})
+}
+
+if(document.readyState === 'complete') {
```
Comment (buunguyen) [inline src/octotree.js]: `$(document).ready(...)` not work? I'm quite surprised if that's the case.

nitpick: need a space after `if`.

File: src/config/edge/manifest.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{
+    "name": "Octotree",
+    "version": "$VERSION",
+    "manifest_version": 2,
+    "author": "Buu Nguyen",
+    "description": "Code tree for GitHub",
+    "icons": {
```
Comment (buunguyen) [inline src/config/edge/manifest.json]: Which manifest file Edge uses? Why icons are defined in both files?

File: src/view.tree.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 class TreeView {
   constructor($dom, store, adapter) {
+    const isEdge = window.navigator.userAgent.indexOf('Edge') > 0;
```
Comment (buunguyen) [inline src/view.tree.js]: Would be great if there's a way to make it work without specific browser detection & handling like this.

File: gulpfile.babel.js [hunk 4]
```diff
@@ -148,6 +148,27 @@ gulp.task('safari', ['safari:js'], () => {
   )
 })
 
+// Edge
+gulp.task('edge:template', () => {
+  return buildTemplate({SUPPORT_FILE_ICONS: true})
```
Comment (buunguyen) [inline gulpfile.babel.js]: You mentioned in the issue file icons aren't working. If that's the case, should mark this as 'false' until there's a fix. That being said, I'd love to have that fixed before merging, unless Edge can't really support fonts.

**Comment by iRYO400** [comment]
Thumb up! Waiting for Edge version!

File: src/config/edge/manifest.json (hunk 2 above)
Comment (Jack-Q) [reply src/config/edge/manifest.json]: Actually, Edge extension is packaged as `appx` and published to Microsoft Store. Both manifests are used.

* `manifest.json` is used by Edge and the icon is displayed within Edge;
* `AppxManifest.xml` is used by Microsoft Store and the icon is displayed in the Marketplace.

They should be the same icon but size requirement of them are not the same.

File: src/octotree.js (hunk 1 above)
Comment (Jack-Q) [reply src/octotree.js]: I don't know the reason why `$(document).ready(...)` fails either.

File: src/view.tree.js (hunk 3 above)
Comment (Jack-Q) [reply src/view.tree.js]: It seems that this problem is resolved in the latest version of Edge. I have removed the browser detection code.

File: gulpfile.babel.js (hunk 4 above)
Comment (Jack-Q) [reply gulpfile.babel.js]: The support configuration is removed for Edge. Actually, this is caused by an issue impacting how Edge handles Content Security Policy headers, which blocks the loading of icon fonts. This issue is fixed but not flighted yet. (Related Issue: [https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/11320212/]())

**Comment by jamesrweb** [comment]
@buunguyen @Jack-Q any movement upcoming to this, would be good to get things merged if all minor topics (seem to be) resolved?

**Comment by piranna** [comment]
Microsoft Edge has got deprecated, I think this can be rejected.

**Comment by abulka** [comment]
Now that Microsoft Edge has been reborn with a chrome engine - is there any reason why Octotree isn't available as an extension for Microsoft Edge?

**Comment by buunguyen** [comment]
Edge version: https://microsoftedge.microsoft.com/addons/detail/octotree/joagmknfcgpikbadjkaikmnhpjadihjg?hl=en-US

**Pull request closed without merging**